src/ServerRequestFactory.php

```diff
@@ -132,29 +153,47 @@ private function getUri(array $server): UriInterface
         return $uri;
     }
 
-    /**
-     * @suppress PhanUndeclaredFunction
-     */
     private function getHeadersFromGlobals(): array
     {
-        if (\function_exists('\getallheaders')) {
-            $headers = \getallheaders();
-            if ($headers === false) {
-                $headers = [];
+        if (function_exists('getallheaders')) {
+            $headers = getallheaders();
+            return $headers === false ? [] : array_filter($headers, static fn (string $value): bool => $value !== '');
```

Comment:
Is it possible to get a empty header value here?

Reply:
Yes.

Reply:
> Yes.

And why is this case incorrect?
https://tools.ietf.org/html/rfc7230#section-5.4
In some cases the Host header can be empty but this header is required

Reply:
I agree, need to add a condition for the host here.

Reply:
or.. for all headers?
https://tools.ietf.org/html/rfc7230#section-3.2
```
header-field   = field-name ":" OWS field-value OWS
field-value    = *( field-content / obs-fold )
```